src/InteractiveWindow/Editor/InteractiveWindow.cs

```diff
@@ -513,56 +401,9 @@ private static void DumpCmd(string prefix, int result, ref Guid pguidCmdGroup, u
 
         #region Caret and Cursor
 
-        private ITextCaret Caret => _textView.Caret;
-
         private void CaretPositionChanged(object sender, CaretPositionChangedEventArgs e)
```

Comment:
There doesn't seem to be a guarantee that the event will be raised on the UI thread and marshalling is the outer object's responsibility.